### ISSUE TITLE
Refactor notifications with toasts and modals

### ISSUE THREAD

**Issue by sven1103** [qbicsoftware/data-manager-app#799]
### What are you missing?

**Success notifications (Toasts)**

Usage: Triggered after successful completion of each task

- [ ] Project creation and editing
- [ ] Experiment creation and editing
- [ ] Experimental variables creation and editing
- [ ] Experimental groups creation and editing
- [ ] Offer upload and remove
- [ ] QC upload and remove 
- [ ] PAT token creation, copying and deletion
- [ ] Project Access person add and remove (also email notifications for the same)
- [x] #973
- [ ] Sample batch dialog add, remove row
- [x] #977
- [ ] Measurements - RAW data available 

**Warning notifications (Dialogs)**

Usage: To warn the user of potential risks that may arise during an interaction. 

- [ ] Confirmation msgs on delete Sample batch
- [ ] Confirmation msgs on delete PAT
- [ ] Confirmation msgs on delete measurement

**Information notifications (Dialogs)**
Confirmation msg on dialog cancellation 

- [x] cancel project creation
- [x] cancel project editing
- [x] cancel experiment creation
- [x] cancel experiment editing
- [x] cancel exp. variable creation
- [x] cancel exp. variable editing
- [x] cancel sample batch registration
- [ ] PAT copy 
- [ ]  TSV conversion in dialogs (Register and Edit measurements)

**Error Notifications**

Usage: To inform users of critical issues and guide them to resolution.

- [ ] File upload : exceeding size, improper TSV, technical errors (DB, network, etc)
- [ ] Editing experiment design after samples are registered.
- [ ] Deleting measurements when no item is selected.

### Why do you need a solution?

Depends on #682 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wow-such-code** [comment]
Is cancel confirmation for experimental groups missing by design?

**Comment by Shraddha0903** [comment]
> Is cancel confirmation for experimental groups missing by design?

Yes, we don't have the cancel confirmation for the experimental groups by design as imo the accidental action of the Cancel button is not potentially as harmful as the others. The lost information could cost the users not more than a few minutes to reenter compared to the increased amount of clicks caused by the cancel confirmation that would annoy them.
We should keep them only for the dialogs where the action could be destructive and would cost some time and effort for the user.
We should also remove it for the sample batch registration dialog, as there won't be the spreadsheet component anymore.
